mParticle-Apple-SDK/Next Gen/MParticleKit/MParticleKit.m

```diff
@@ -0,0 +1,192 @@
+//
+//  MParticleKit.m
+//  mParticle-Apple-SDK
+//
+//  Created by Ben Baron on 6/29/23.
+//
+
+#import "MParticleKit.h"
+#import "MPEnums.h"
+#import "MPIConstants.h"
+#import "MPKitContainer.h"
+#import "MPDataPlanFilter.h"
+#import "MPNetworkCommunication.h"
+#import "MParticleWebView.h"
+#import "MPMessage.h"
+#import "MPPersistenceController.h"
+#import "MPStateMachine.h"
+#import "MPUploadBuilder.h"
+#import "MPBackendController.h"
+
+const NSInteger kMParticleKitCode = 999999999;
+
+@interface MParticle()
```

Comment:
Im not sure if this is an implementation outline in progress or the finished version?
I think that the MParticle should almost serve as a facade and maybe have a state. I wouldn't include a low level layer as the persistance in the highest external one, or things like dataPlanId/version (maybe we can create a manager for those)

Reply:
Agreed in the medium/long term, but for this initial change, I was trying to limit the scope as much as possible so I didn't want to make big changes to the mParticle class yet. I actually had pulled a bunch of logic out of it and into a separate class, but it just made the PR harder to grok as it added more noise, so I pulled it out and left it in a side branch for now.